test/e2e/config/vsphere.yaml

```diff
@@ -26,6 +26,15 @@ providers:
   - name: cluster-api
     type: CoreProvider
     versions:
+      - name: "v1.9.99"
+        value: "https://storage.googleapis.com/k8s-staging-cluster-api/components/main/core-components.yaml"
```

Comment:
I think it would be better to pin this to a nightly matching our go dependency to retain control over what we test + avoid that go and image get out of sync